casadm/cas_lib.c

```diff
@@ -2880,3 +2880,78 @@ int check_cache_device(const char *device_path)
 
 	return SUCCESS;
 }
+
+int zero_md(const char *cache_device, uint8_t force, uint16_t cache_id){
+	struct kcas_zero_md cmd;
+	struct kcas_cache_check_device cmd_info;
+	char zero_page[4096] = {0};
+	int fd = 0, caches, i;
+	int *ids;
+
+	/* check if cache device given exists */
+	fd = open(cache_device, O_RDONLY);
+	if (fd < 0) {
+		cas_printf(LOG_ERR, "Device '%s' not found.\n", cache_device);
+		return FAILURE;
+	}
+	close(fd);
+
+	_check_cache_device(cache_device, &cmd_info);
+	if (!cmd_info.is_cache_device) {
+		cas_printf(LOG_ERR, "Device '%s' does not contain OpenCAS's metadata.\n", cache_device);
+		return FAILURE;
+	}
+
+	/* don't even attempt ioctl if cache is in use and without 'force' flag*/
+	if (check_cache_already_added(cache_device) == FAILURE && !force) {
+		cas_printf(LOG_ERR, "Cache device '%s' is already used as cache.\n"
+				"Use 'force' flag if You're pretty sure You want to stop cache and clear metadata.\n",
+				cache_device);
+		return FAILURE;
+	}
+
+	caches = get_cache_count(fd);
+	ids = get_cache_ids(&caches);
+
+	memset(&cmd, 0, sizeof(cmd));
+
+	if (cache_id) {
+		for (i = 0; i < caches; i++) {
+			if (cache_id == ids[i])
+				cmd.cache_id = cache_id;
+		}
+		if (!cmd.cache_id) {
+			cas_printf(LOG_NOTICE, "Cache with ID %d doesn't exist.\n", cache_id);
+			return FAILURE;
+		}
+	}
+
+	strncpy_s(cmd.device_path_name,
+		sizeof(cmd.device_path_name),
+		cache_device,
+		strnlen_s(cache_device,
+			sizeof(cmd.device_path_name)));
+
+	cmd.force = force;
+
+	if(cmd.cache_id && cmd.force) {
+		stop_cache(cmd.cache_id, 0);
```

Comment:
I think that stopping cache is not objective of this command. Let's keep it simple - user should stop cache prior to executing this command.